test/http-basic-test.js

```diff
@@ -116,6 +116,31 @@ test('success: no response', () => {
   })
 })
 
+test('success: no response (conditional)', () => {
+  const server = http.createServer((req, res) => {
+    serverTiming({
+      enabled: req => req.url.includes('?debug=true')
+    })(req, res)
+    res.setMetric('foo', 100.0)
+    res.end('hello')
+  }).listen(0, () => {
+    http.get(`http://localhost:${server.address().port}/?debug=true`, mustCall((res) => {
+      let assertStream = new AssertStream()
+      assertStream.expect('hello')
+      res.pipe(assertStream)
+      assert(res.headers['server-timing'])
+
+      http.get(`http://localhost:${server.address().port}/?debug=false`, mustCall((res) => {
+        assertStream = new AssertStream()
```

Comment:
we would be better to set `const` signature, please do not re-use same variable.

Reply:
Fair enough. Fixed.